Update reserved names

KellerFuchs opened the pull request:
- [x] Reserve RFC2142, services and email aliases names
- [x] Make the list more readable and move it to a separate file
- [x] Automatically pluralize names

KellerFuchs committed:
utils: Add and split out RFC2142 names

KellerFuchs committed:
utils.py: Add #! email aliases

KellerFuchs committed:
utils: Reserve names that are already #! services

KellerFuchs committed:
utils: Comment and sort the remaining reserved names

KellerFuchs committed:
utils: Automatically pluralize reserved names

KellerFuchs committed:
Split off the list of reserved usernames in reserved.py

At the same time, make it a `frozenset` to prevent accidental
modification of the list (and possibly slightly-faster lookups)

KellerFuchs committed:
utils/validate_username: Update link to hashbang.sh

I'm not sure what would be a better option here...

KellerFuchs committed:
Get rid of password-related functionality

We aren't using it anyhow, and `make_salt` was terrifying

singlerider commented on provisor/reserved.py:
```diff
@@ -0,0 +1,35 @@
+reserved_usernames = [
+    # Names that might be used for fishing
```
Phishing?

KellerFuchs replied:
Fixed.

KellerFuchs committed:
utils: validate_username: Don't test whether the user exists

`pwd` interrogates the local NSS database, which might not be tied to LDAP.

It's the case on the container running the website,
which anyways handles the case where the user exists in LDAP

singlerider commented on provisor/reserved.py:
```diff
@@ -0,0 +1,35 @@
+    'uucp', 'postmaster', 'webmaster', 'www'
+]
+
+reserved_usernames += [ name + 's'
```
Not Pep8 compliant

KellerFuchs replied:
OK, added a commit, and `pep8` runs cleanly on `reserved.py`.
(And I'm so bloody not making the rest PEP8-compliant in this PR)

singlerider commented on provisor/utils.py:
```diff
@@ -67,29 +67,22 @@ def validate_pubkey(value):
+    if value in RESERVED_USERNAMES:
         raise ValueError('Username is reserved')
+
     user_exists = True
```
Why not include `value` in the error message?

Up to you; just a suggestion.

KellerFuchs replied:
Then I would need to uniformize with the stuff in `hashbang.sh`, and that's kinda out-of-scope for this PR anyhow.

singlerider commented on provisor/utils.py:
```diff
@@ -67,29 +67,22 @@ def validate_pubkey(value):
         pwd.getpwnam(value)
     except:
         user_exists = False
+
```
`except` what?
I know this isn't part of the PR, but it's staring me in the face.
Implicit Exception handling ain't cool.

KellerFuchs replied:
I have another branch which removes this code anyhow  ;)

KellerFuchs committed:
utils: Use local imports

This way, it is clear which functions perform OS interactions

singlerider reviewed (changes requested):
At the very least, PEP8 should be adhered to.

KellerFuchs committed:
reserved: Add daemon usernames from Debian

KellerFuchs committed:
reserved: Make the file comply with PEP8

This is just to make @singlerider happy,
nothing else in this codebase is PEP8...

KellerFuchs committed:
Go phish.

Where I learn I don't know how to spell

singlerider commented:
I'll open a PR as soon as this is merged to PEP8 all of this.

KellerFuchs committed:
tox: Allow multiple imports on a line

KellerFuchs committed:
utils: validate_pubkey: Avoid bare except

KellerFuchs commented:
@singlerider If you could do the signed merge, that would be super-convenient.

KellerFuchs committed:
Merge branch 'cleanup' into reserved-names
